src/runtime/hexagon_host.cpp

```diff
@@ -591,67 +583,66 @@ WEAK int halide_hexagon_copy_to_host(void *user_context, buffer_t* buf) {
     return 0;
 }
 
-WEAK int halide_hexagon_device_sync(void *user_context, struct buffer_t *) {
+WEAK int halide_hexagon_device_sync(void *user_context, struct halide_buffer_t *) {
     debug(user_context)
         << "Hexagon: halide_hexagon_device_sync (user_context: " << user_context << ")\n";
     // Nothing to do.
     return 0;
 }
 
-WEAK int halide_hexagon_wrap_device_handle(void *user_context, struct buffer_t *buf,
+WEAK int halide_hexagon_wrap_device_handle(void *user_context, struct halide_buffer_t *buf,
                                            void *ion_buf, uint64_t size) {
-    halide_assert(user_context, buf->dev == 0);
-    if (buf->dev != 0) {
+    halide_assert(user_context, buf->device == 0);
+    if (buf->device != 0) {
         return -2;
     }
 
-    ion_device_handle *handle = (ion_device_handle*)  halide_malloc(user_context, sizeof(ion_device_handle));
+    ion_device_handle *handle = (ion_device_handle *)malloc(sizeof(ion_device_handle));
```

Comment:
For small allocations like this with no special alignment requirements I think we should use malloc (that's what we do with device wrappers now). It's lower overhead.